functional_widget/test/diagnostics_test.dart

```diff
@@ -125,6 +125,27 @@ class FunctionTest extends StatelessWidget {
   @override
   Widget build(BuildContext _context) => functionTest(a);
   @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<void Function()>('a', a));
+  }
+}
+'''));
+    });
+    test('typedef type', () async {
+      // @todo should be `final void Function(T) a;` instead of
+      // `final void Function(dynamic) a;`
+      // and `DiagnosticsProperty<void Function(T)>` instead of
+      // `ObjectFlagProperty<dynamic>.has`
+      await _expect('typedefTest', completion('''
+class TypedefTest<T> extends StatelessWidget {
+  const TypedefTest(this.a, {Key key}) : super(key: key);
+
+  final void Function(dynamic) a;
+
+  @override
+  Widget build(BuildContext _context) => typedefTest<T>(a);
+  @override
```

Comment:
functionally seems to be suboptimal, but at least for now it's still working for developer, and not crash due to https://github.com/rrousselGit/functional_widget/pull/67/files#diff-8725693a4d1b56a9253011b227f468ddL7 `element.session` being `null`.